src/hooks/businessPersonnel/useUpdateBusinessPersonnel.ts

```diff
@@ -0,0 +1,102 @@
+import useSWRMutation from 'swr/mutation'
+import { post } from '../../api/layer/authenticated_http'
+import type { BusinessPersonnel, PersonnelRole, RawBusinessPersonnel } from './types'
+import { useAuth } from '../useAuth'
+import { useLayerContext } from '../../contexts/LayerContext'
+import { useSWRConfig } from 'swr'
+import { withSWRKeyTags } from '../../utils/swr/withSWRKeyTags'
+import { BUSINESS_PERSONNEL_TAG_KEY } from './useBusinessPersonnel'
+
+type UpdateBusinessPersonnelBody = {
+  id: string
+} & Partial<
+  Pick<
+    RawBusinessPersonnel,
+    'full_name' | 'preferred_name' | 'external_id'
+  > & {
+    email_addresses: ReadonlyArray<{ email_address: string }>
+    phone_numbers: ReadonlyArray<{ phone_number: string }>
+    roles: ReadonlyArray<{ role: PersonnelRole }>
+  }
+>
+
+const updateBusinessPersonnel = post<
```

Comment:
Usually, we have this in the `src/api/*`, and I like that approach because of having all used API endpoints in one place.